src/GraphQL/Wrappers/Response/ViolationResponse.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\graphql\GraphQL\Wrappers\Response;
+
+use Drupal\graphql\GraphQL\Wrappers\Violation\ViolationCollection;
+
+/**
+ * Base class for responses containing the violations.
+ */
+class ViolationResponse implements ResponseInterface {
+
+  /**
+   * Violations.
+   *
+   * @var \Drupal\graphql\GraphQL\Wrappers\Violation\ViolationCollection
+   */
+  protected $violations;
+
+  /**
+   * ViolationResponse constructor.
+   *
+   * @param \Drupal\graphql\GraphQL\Wrappers\Violation\ViolationCollection|null $violations
+   *   List of violations occurred during query or mutation.
+   */
+  public function __construct(?ViolationCollection $violations = NULL) {
+    $this->violations = $violations ?: new ViolationCollection();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function code(): int {
+    return 200;
+  }
+
+  /**
+   * Gets the violations.
+   *
+   * @return array
+   *   List of violations occurred during query or mutation.
+   */
+  public function errors(): array {
```

Comment:
why is this not part of the interface?

Reply:
This is because not every response needs to necessarily use violations. At least that's how it is in our code. For example we have `RedirectResponse` and it's not printing any violations. It just returns a target url/path. But maybe it would be still okay to put it on interface. Let's try it and we'll see